src/App.tsx

```diff
@@ -1,61 +1,133 @@
 import './App.scss';
+import { useState, useEffect } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { User } from './components/types/User';
+import { Todo } from './components/types/Todo';
+
+import { TodoList } from './components/TodoList/TodoList';
 
 export const App = () => {
+  const [copyTodos, setCopyTodos] = useState([...todosFromServer]);
+  const [selectedUserName, setSelectedUserName] = useState('');
+  const [newTitle, setNewTitle] = useState('');
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+
+  const maxId = Math.max(...todosFromServer.map(todo => +todo.id));
+  const findUser = (userName: string): User | null => {
+    return usersFromServer.find(user => user.name === userName) || null;
+  };
+
+  const checkError = () => {
+    if (!newTitle) {
+      setErrorTitle(true);
+    }
+
+    if (!selectedUserName) {
+      setErrorUser(true);
+    }
+  };
+
+  const createTodo = () => {
+    const user = findUser(selectedUserName);
+
+    if (user && newTitle) {
+      const newTodo = {
+        id: maxId,
+        title: newTitle,
+        completed: false,
+        userId: user.id,
+      };
+
+      setCopyTodos(todos => [...todos, newTodo]);
+      setNewTitle('');
+      setSelectedUserName('');
+    }
+
+    return null;
+  };
+
+  const addHandler
+  = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    event.preventDefault();
+    createTodo();
+    checkError();
+  };
+
+  const todos: Todo[] = copyTodos.map(todo => {
+    return {
+      ...todo,
+      user: usersFromServer.find(user => todo.userId === user.id),
+    };
+  });
+
+  useEffect(() => {
+    if (newTitle) {
```

Comment:
The 'useEffect' hook is missing its dependency array. This could lead to unnecessary renders or stale data. It should include 'newTitle' and 'selectedUserName'.